platform-sdk/docs/proposals/inlinePces/inlinePcesProposal.md

```diff
@@ -0,0 +1,203 @@
+> file system rather than a conventional, file storage system. Recent changes to the MappedByteBuffer API mean that it
+> supports all the behaviours needed to allow direct memory updates and provide the durability guarantees needed for
+> higher level, Java client libraries to implement persistent data types (e.g. block file systems, journaled logs,
+> persistent objects, etc.). The implementations of FileChannel and MappedByteBuffer need revising to be aware of this
+> new backing type for the mapped file.
+
+However, it also limits support for these more rigorous semantics to Linux:
+
+> The target OS/CPU platform combinations for this JEP are Linux/x64 and Linux/AArch64. This restriction is imposed for
+> two reasons. This feature will only work on OSes that support the mmap system call MAP_SYNC flag, which allows
+> synchronous mapping of non-volatile memory. That is true of recent Linux releases. It will also only work on CPUs that
+> support cache line writeback under user space control. x64 and AArch64 both provide instructions meeting this
+> requirement.
+
+Only Linux is used today in production deployments. Should MacOS or Windows be used in the future, we may need to look
+into equivalent solutions for those platforms, or use something other than memory mapped files on those platforms, with
+a corresponding impact on time to finality for those hosts.
+
+It's plausible that we may be able to get the main implementation functional on other OSes (perhaps with reduced
+consistency guarantees). If this is not plausible, an implementation that does not use memory mapped files will be
+developed for use on MacOS and Windows. The goal of this solution will be to allow developers to run the software
+locally. Since this will not be code we intend to deploy (in the short term), we will just need to make it
+"fast enough" such that local tests work as intended. It will not be necessary to fine tune the performance of this
+implementation until we derive a use case that requires it.
+
+The [Man Page for mmap(2)](https://man7.org/linux/man-pages/man2/mmap.2.html) on Linux describes the MAP_SYNC flag:
+
+> MAP_SYNC (since Linux 4.15)
+>    This flag is available only with the MAP_SHARED_VALIDATE
+>    mapping type; mappings of type MAP_SHARED will silently
+>    ignore this flag.  This flag is supported only for files
+>    supporting DAX (direct mapping of persistent memory).  For
+>    other files, creating a mapping with this flag results in
+>    an EOPNOTSUPP error.
+>
+>    Shared file mappings with this flag provide the guarantee
+>    that while some memory is mapped writable in the address
+>    space of the process, it will be visible in the same file
+>    at the same offset even after the system crashes or is
+>    rebooted.  In conjunction with the use of appropriate CPU
+>    instructions, this provides users of such mappings with a
+>    more efficient way of making data modifications
+>    persistent.
+
+### Performance
```

Comment:
does PCES have recovery mechanisms for IO failures? Now that synchronous write to disk will be in the way of consensus (gossip will have to wait for durability of self-events), how will we handle if write doesn't succeed?

Reply:
The only other thing i'd add is could you please enrich the title a bit? Sounds quite different from the description of the proposal.